hydra_provider/src/data_feed_service/random/iterator.rs

```diff
@@ -0,0 +1,180 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::SystemTime;
+use rand::Rng;
+use rand::{rngs::StdRng, SeedableRng};
+
+use messages::{command::Command, health::Health, sensor::{Air, EkfNav1, EkfNav2, EkfQuat, GpsPos1, GpsPos2, GpsVel, Imu1, Imu2, Sensor, UtcTime}, state::{State, StateData}, Data, Log, Message};
+
+
+pub struct RandomDataFeedIterator {
+	pub is_running: Arc<AtomicBool>,
+}
+
+impl Iterator for RandomDataFeedIterator {
+	type Item = Message;
+
+	fn next(&mut self) -> Option<Message> {
+		if !self.is_running.load(Ordering::Relaxed) {
+			return None
+		}
```

Comment:
Is `is_running` blocking? If not take a peek at.

https://doc.rust-lang.org/std/sync/struct.Condvar.html

Reply:
Sounds good!